android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/AnalyticsViewModel.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.view.viewmodel
+
+import android.view.View
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import br.com.zup.beagle.newanalytics.AnalyticsService
+import br.com.zup.beagle.android.action.ActionAnalytics
+import br.com.zup.beagle.android.widget.RootView
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+internal class AnalyticsViewModel : ViewModel() {
+
+    fun createActionReport(
+        rootView: RootView,
+        origin: View,
+        action: ActionAnalytics,
+        analyticsValue: String?,
+    ) {
+        viewModelScope.launch(Dispatchers.Default) {
```

Comment:
This context can be ommited
```suggestion
        viewModelScope.launch() {
```

Reply:
I'm not sure if this context can be omitted.

`viewModelScope` runs on the `Dispatchers.Main by` default. **This is the UI thread**. It seems to me that `AnalyticsService.createActionRecord` could be very resource consuming due to the fact that it uses reflection in it's internal implementation. That said, I think that `Dispatchers.Default` is the best choice to run this job.